plugins/meta/portmap/chain_test.go

```diff
@@ -195,4 +198,48 @@ var _ = Describe("chain tests", func() {
 			}
 		}
 	})
+
+	It("deletes chains idempotently in parallel", func() {
+		defer cleanup()
+		// number of parallel executions
+		N := 10
+		var wg sync.WaitGroup
+		err := testChain.setup(ipt)
+		Expect(err).NotTo(HaveOccurred())
+		errCh := make(chan error, N)
+		for i := 0; i < N; i++ {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				// Use the testing namespace
+				runtime.LockOSThread()
+				defer runtime.UnlockOSThread()
+				err = testNs.Set()
```

Comment:
Test has already been in test namespace after `BeforeEach`.

Reply:
yeah, and it goes back to it because of L67.
The problem is that the new goroutines use the root namespace

Reply:
L67 in which file? I'm sorry I didn't find it, could you help to paste a link here?

Reply:
in L67 of this file there is a `runtime.LockOSThread()` so the test is always in that namespace, but I think that your question is why I have to set again the namespace inside the goroutine.
I'm doing it because is the only way I could make that the goroutine uses the testing namespace, otherwise they always use the root namespace, don't know if there is a better way though

Reply:
Yes, as of go 1.10 new goroutines always spawn in the host netns. However, this should be a testNs.Do( closure ) to ensure we *leave* the testNS!